From 327eddb7010a5dce8764475c492cc40fea6b9b05 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Fri, 21 Aug 2020 14:32:14 +0800
Subject: [PATCH] drm: Remove global property HDR_PANEL_METADATA

HDR_PANEL_METADATA has been hdmi private property, so remove
HDR_PANEL_METADATA in mode_config.

Fixes: 97fd2e42b6da ("drm: Add HDR panel metadata property")
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Change-Id: Ia534d61d78edcdd2260b41d1c5a076f19034bf75
---
 drivers/gpu/drm/drm_connector.c | 30 ------------------------------
 include/drm/drm_connector.h     |  3 ---
 include/drm/drm_mode_config.h   |  2 --
 3 files changed, 35 deletions(-)

diff --git a/drivers/gpu/drm/drm_connector.c b/drivers/gpu/drm/drm_connector.c
index 688b37ee24f4..5f1a4d0c6f6d 100644
--- a/drivers/gpu/drm/drm_connector.c
+++ b/drivers/gpu/drm/drm_connector.c
@@ -1026,14 +1026,6 @@ int drm_connector_create_standard_properties(struct drm_device *dev)
 		return -ENOMEM;
 	dev->mode_config.hdr_output_metadata_property = prop;
 
-	prop = drm_property_create(dev,
-				   DRM_MODE_PROP_BLOB |
-				   DRM_MODE_PROP_IMMUTABLE,
-				   "HDR_PANEL_METADATA", 0);
-	if (!prop)
-		return -ENOMEM;
-	dev->mode_config.hdr_panel_metadata_property = prop;
-
 	return 0;
 }
 
@@ -1583,28 +1575,6 @@ int drm_connector_update_edid_property(struct drm_connector *connector,
 }
 EXPORT_SYMBOL(drm_connector_update_edid_property);
 
-int
-drm_mode_connector_update_hdr_property(struct drm_connector *connector,
-				       const struct hdr_static_metadata *data)
-{
-	struct drm_device *dev = connector->dev;
-	size_t size = sizeof(*data);
-	struct drm_property *property =
-			dev->mode_config.hdr_panel_metadata_property;
-	struct drm_property_blob **replace =
-		&connector->state->hdr_panel_blob_ptr;
-	int ret;
-
-	ret = drm_property_replace_global_blob(dev,
-					       replace,
-					       size,
-					       data,
-					       &connector->base,
-					       property);
-	return ret;
-}
-EXPORT_SYMBOL(drm_mode_connector_update_hdr_property);
-
 /**
  * drm_connector_set_link_status_property - Set link status property of a connector
  * @connector: drm connector
diff --git a/include/drm/drm_connector.h b/include/drm/drm_connector.h
index 281ea0fc59c4..f7b56c31deff 100644
--- a/include/drm/drm_connector.h
+++ b/include/drm/drm_connector.h
@@ -1313,9 +1313,6 @@ int drm_connector_set_path_property(struct drm_connector *connector,
 int drm_connector_set_tile_property(struct drm_connector *connector);
 int drm_connector_update_edid_property(struct drm_connector *connector,
 				       const struct edid *edid);
-int
-drm_mode_connector_update_hdr_property(struct drm_connector *connector,
-				       const struct hdr_static_metadata *data);
 void drm_connector_set_link_status_property(struct drm_connector *connector,
 					    uint64_t link_status);
 int drm_connector_init_panel_orientation_property(
diff --git a/include/drm/drm_mode_config.h b/include/drm/drm_mode_config.h
index 6ef792317cfc..adba21d96f8b 100644
--- a/include/drm/drm_mode_config.h
+++ b/include/drm/drm_mode_config.h
@@ -814,8 +814,6 @@ struct drm_mode_config {
 	 */
 	struct drm_property *hdr_output_metadata_property;
 
-	struct drm_property *hdr_panel_metadata_property;
-
 	/* dumb ioctl parameters */
 	uint32_t preferred_depth, prefer_shadow;
 
-- 
2.35.3

